doc/adr/0010-atomic-composite-grapth-operations.md

```diff
@@ -0,0 +1,267 @@
+    - Like `:block/save` when new link is discovered, should produce also `:page/create` event
+
+Atomic events should follow same Common Events model as it's happening now.
+
+List of ((⚛️ Atomic Graph Ops))
+- *Page Ops*
+    - ((⚛️**`:page/new`**))
+    - ((⚛️**`:page/rename`**))
+    - ((⚛️**`:page/merge`**))
+    - ((⚛️**`:page/remove`**))
+- *Block Ops*
+    - ((⚛️**`:block/new`**))
+    - ((⚛️**`:block/save`**))
+    - ((⚛️**`:block/open`**))
+    - ((⚛️**`:block/move`** ⭐️))
+    - ⚛️**`:block/remove`**
+- *Shortcut Ops*
+    - (({{[[TODO]]}} ⚛️**`:shortcut/new`**))
+    - (({{[[TODO]]}} ⚛️**`:shortcut/remove`**))
+    - (({{[[TODO]]}} ⚛️**`:shortcut/move`**))
+
+
+((⎄ Composite Graph Ops)) are a way to group ((⚛️ Atomic Graph Ops)).
+
+We'll only need one type of Composite Graph Ops, that is Consequence Event.
+
+Consequence Event is 2 things:
+- Trigger Event (like `:paste`)
+- Consequence Events List
+    - Containing however many events to include into transaction resolution.
+    - These events can be more of Consequence Events, or ((⚛️ Atomic Graph Ops)).
+
+
+This will allow for easier Temporality,
+We'll be able to represent Main Action that moved Knowledge Graph forward.
+
+It will maintain information about `:paste` events that injected sub-tree with ((⚛️ Atomic Graph Ops)).
+
+## Consequences
+
+We'll have additional work of Porting Semantic Events to use Atomic Graph Ops and compositions.
+
+We have working `:paste` & `:block/save` events, they are broken not ported now.
+
+We'll have smaller amount of tests in order to provide correctness guarantees
+
+
+## Additional Resources
+
+### Catalog of operations
+
+- Types of Graph Operations
+    - ⚛️ Atomic Graph Ops
+        - Not divisible Graph Ops
+        - Operations like create new block, create page, save block
+    - ⎄ Composite Graph Ops
+        - Collection of events to be executed on the graph
+        - Like `:block/save` when new link is discovered, should produce also `:page/create` event
+
+- List of Atomic Graph Operations:
+    - Page Ops
+        - ⚛️**`:page/new`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `title` - Page title page to be created
+                - `page-uid` - `:block/uid` of page to be created
+                - `block-uid` - `:block/uid` of 1st block to be created in page to be created
+        - ⚛️**`:page/rename`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `page-uid` - `:block/uid` of page to be renamed
+                - `new-name` - Page should have this name after operation
+                - `old-name` - ^^To Remove^^ This is accidental, and shouldn't be provided
+        - ⚛️**`:page/merge`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `page-uid` - `:block/uid` of page to be merged into `new-page`
+                - `new-page` - page name of a page we'll merge contents of `page-uid` page into
```

Comment:
This should be `target-page-uid`, I think. It's a bit weird how page identity seems split between name and uid. Some operations take both because they really needed, but if they just need to identify the page then there should be a single unique identifier, and that sounds like the uid.